src/Thorn_Calculus-Core_Bisimilarities.thy

```diff
@@ -1469,9 +1467,177 @@ proof -
   finally show ?thesis .
 qed
 
+private lemma adapted_after_repeated_receive_nested_idempotency:
```

Comment:
Where `_after_` is used in other lemma names, these lemma names have the structure `⟨function_1⟩_after_⟨function_2⟩`. They refer to the left-hand sides of the stated bisimilarities, which arise from applying `⟨function_2⟩` and then `⟨function_1⟩`. However, in the case of this lemma and the next, what comes after `_after_` is a lemma name.

Therefore, the name of this lemma and the name of the next should be changed. Please look whether we have other lemmas in this theory that state bisimilarities to be used in “up to bisimilarity” proofs and, if yes, use the same naming scheme as used for those lemmas.

Reply:
The following are the names of the lemmas in this theory that state bisimilarities to be used in “up to bisimilarity” proofs:

- `create_channel_power_after_move_adapted`
- `independent_value_adjustment`
- `communication_with_rightmost_adjustment`

I'm not sure whether there is a concrete naming scheme used for those lemmas beyond what we have already agreed upon. Therefore, since our two auxiliary lemmas are essentially versions of `repeated_receive_nested_idempotency` and `parallel_left_commutativity` with adaptations (and other features, of course), the following are my suggestions for the new names:

- `adapted_repeated_receive_nested_idempotency` instead of `adapted_after_repeated_receive_nested_idempotency`
- `adapted_parallel_left_commutativity` instead of `adapted_after_parallel_left_commutativity`

What do you think of these new names?

Reply:
Worst of all, I have no idea why I chose the name `communication_with_rightmost_adjustment`. I have no idea what this is supposed to mean. :sweat_smile:

Reply:
I think the problem with the new names proposed by you is exactly what you wrote in parentheses: “and other features, of course”. :slightly_smiling_face: Having lemma names that ignore such other features doesn’t seem good to me, as such names would be misleading.

I’m thinking about going back to having the lemma names describe the left-hand sides of the bisimilarities, like it is the case with many other lemmas in the Þ-calculus formalization. There are two challenges with this approach:

  * The second lemma has the simpler term on the left and the more complex one on the right, which means that describing the left-hand side doesn’t suggest what the whole lemma states, because the right-hand side is not the “obvious” simplification of the left-hand side, but rather a “complification”.

  * The terms involved are very complex and thus hard to describe verbally.

I propose the following solutions to these problems:

  * We swap the sides of the second lemma’s statement and use the second lemma with the `symmetric` attribute to compensate for that.

  * We don’t describe the structure of the terms but how they arose from the sides of the `inner_repeated_receive_redundancy` statement. Concretely, we call the first lemma `with_inner_repeated_receive_post_right_receive` and the second lemma `without_inner_repeated_receive_post_right_receive`.

What do you think about this idea?

Reply:
The idea sounds sensible indeed.

Reply:
Fixed in 782f8f31672ad6f0b4dd8d6c0f64b1b4b71bcfc8 and 3e1e5dce18ac62324aeda8e4dc8cb28cd2fb4545, please verify.

Reply:
Looks very good.